src/Env.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { FormHelper } from "cockpit-components-form-helper.jsx";
+import { Grid } from "@patternfly/react-core/dist/esm/layouts/Grid";
+import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput";
+import { TrashIcon } from '@patternfly/react-icons';
+import cockpit from 'cockpit';
+
+import * as utils from './util.js';
+
+const _ = cockpit.gettext;
+
+export function validateEnvVar(env, key, validationFailed) {
+    const delta = { ...validationFailed };
+
+    switch (key) {
+    case "envKey":
+        if (utils.isEmpty(env.envKey))
+            delta.envKey = _("Key must not be empty");
+        break;
+    case "envValue":
+        if (utils.isEmpty(env.envValue))
+            delta.envValue = _("Value must not be empty");
+
+        break;
+    }
+
+    return delta;
+}
+
+const handleEnvValue = (key, value, idx, onChange, additem, itemCount, companionField) => {
+    // Allow the input of KEY=VALUE separated value pairs for bulk import only if the other
+    // field is not empty.
+    if (value.includes('=') && !companionField) {
+        const parts = value.trim().split(" ");
+        let index = idx;
+        for (const part of parts) {
+            const [envKey, ...envVar] = part.split('=');
+            if (!envKey || !envVar) {
+                continue;
+            }
+
+            if (index !== idx) {
+                additem();
+            }
+            onChange(index, 'envKey', envKey);
+            onChange(index, 'envValue', envVar.join('='));
+            index++;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20230907-093411-a9f37bd4-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)